x/dsmr/block.go

```diff
@@ -1,69 +1,65 @@
 // Copyright (C) 2024, Ava Labs, Inc. All rights reserved.
 // See the file LICENSE for licensing terms.
 
+//go:generate go run github.com/StephenButtolph/canoto/canoto --concurrent=false $GOFILE
+
 package dsmr
 
 import (
+	"github.com/StephenButtolph/canoto"
 	"github.com/ava-labs/avalanchego/ids"
 	"github.com/ava-labs/avalanchego/utils/crypto/bls"
-	"github.com/ava-labs/avalanchego/utils/wrappers"
 	"github.com/ava-labs/avalanchego/vms/platformvm/warp"
 
 	"github.com/ava-labs/hypersdk/codec"
-	"github.com/ava-labs/hypersdk/consts"
 	"github.com/ava-labs/hypersdk/utils"
 )
 
 const InitialChunkSize = 250 * 1024
 
-type Tx interface {
+type Tx[T any] interface {
+	canoto.FieldMaker[T]
```

Comment:
This is basically the only real change. Rather than expecting the tx to be serializable with the avalanchego codec (which is an undocumented assumption) the interface now specifies canoto.

I had talked with aaron super briefly, but we may want to replace these with `[]byte` so that the transactions can specify their own encoding.